simplified-viewer-preview/src/main/java/org/nypl/simplified/viewer/preview/BookPreviewEmbeddedFragment.kt

```diff
@@ -97,14 +102,40 @@ class BookPreviewEmbeddedFragment : Fragment() {
   }
 
   private fun pauseWebViewPlayer() {
-    webView.loadUrl(
-      "javascript:(function(){" +
-        "button=document.getElementsByClassName('playback-toggle halo')[0];" +
-        "event=document.createEvent('HTMLEvents');" +
-        "event.initEvent('click',true,true);" +
-        "button.dispatchEvent(event);" +
-        "})()"
-    )
+    webView.evaluateJavascript("pausePlayer()", null)
+  }
+
+  private inner class CustomWebViewClient : WebViewClient() {
+    override fun onPageFinished(view: WebView?, url: String?) {
+      injectJavaScript()
+      super.onPageFinished(view, url)
+    }
+
+    private fun injectJavaScript() {
+      try {
+        val inputStream = requireActivity().assets.open("preview_player_commands.js")
+        val buffer = ByteArray(inputStream.available())
+        inputStream.read(buffer)
+        inputStream.close()
+
+        val encodedContent = URLEncoder.encode(String(buffer, Charset.forName(UTF_8)), UTF_8)
+          .replace("+", "%20")
+        val encodedString = Base64.encodeToString(encodedContent.toByteArray(), Base64.NO_WRAP)
```

Comment:
Is there something you can do with [WebViewAssetLoader](https://developer.android.com/develop/ui/views/layout/webapps/load-local-content) to make this more elegant? It would be nice to do `<script src="{url to load file from assets}" />` instead of all this. This is ok if that doesn't work, but I wanted to make sure you tried it.